internal/integrationtest/core/core_test.go

```diff
@@ -0,0 +1,121 @@
+	"go.bug.st/testsuite/requirejson"
+)
+
+func TestCoreSearch(t *testing.T) {
+	env := testsuite.NewEnvironment(t)
+	defer env.CleanUp()
+
+	cli := integrationtest.NewArduinoCliWithinEnvironment(env, &integrationtest.ArduinoCLIConfig{
+		ArduinoCLIPath:         paths.New("..", "..", "..", "arduino-cli"),
+		UseSharedStagingFolder: true,
+	})
+
+	// Set up an http server to serve our custom index file
+	test_index := paths.New("..", "testdata", "test_index.json")
+	url := env.HTTPServeFile(8000, test_index)
+
+	// Run update-index with our test index
+	_, _, err := cli.Run("core", "update-index", "--additional-urls="+url.String())
+	require.NoError(t, err)
+
+	// Search a specific core
+	out, _, err := cli.Run("core", "search", "avr")
+	require.NoError(t, err)
+	require.Greater(t, len(strings.Split(string(out), "\n")), 2)
+
+	out, _, err = cli.Run("core", "search", "avr", "--format", "json")
+	require.NoError(t, err)
+	requirejson.NotEmpty(t, out)
+
+	// additional URL
+	out, _, err = cli.Run("core", "search", "test_core", "--format", "json", "--additional-urls="+url.String())
+	require.NoError(t, err)
+	requirejson.Len(t, out, 1)
+
+	// show all versions
+	out, _, err = cli.Run("core", "search", "test_core", "--all", "--format", "json", "--additional-urls="+url.String())
+	require.NoError(t, err)
+	requirejson.Len(t, out, 2)
+	// requirejson.Len(t, out, 3) // Test failure
+
+	checkPlatformIsInJSONOutput := func(stdout []byte, id, version string) {
+		jqquery := fmt.Sprintf(`[{id:"%s", latest:"%s"}]`, id, version)
+		requirejson.Contains(t, out, jqquery, "platform %s@%s is missing from the output", id, version)
+	}
+
+	// Search all Retrokit platforms
+	out, _, err = cli.Run("core", "search", "retrokit", "--all", "--additional-urls="+url.String(), "--format", "json")
+	require.NoError(t, err)
+	checkPlatformIsInJSONOutput(out, "Retrokits-RK002:arm", "1.0.5")
+	checkPlatformIsInJSONOutput(out, "Retrokits-RK002:arm", "1.0.6")
+	//checkPlatformIsInJSONOutput(out, "Retrokits-RK002:arm", "1.0.9") // Test failure
+
+	// Search using Retrokit Package Maintainer
+	out, _, err = cli.Run("core", "search", "Retrokits-RK002", "--all", "--additional-urls="+url.String(), "--format", "json")
+	require.NoError(t, err)
+	checkPlatformIsInJSONOutput(out, "Retrokits-RK002:arm", "1.0.5")
+	checkPlatformIsInJSONOutput(out, "Retrokits-RK002:arm", "1.0.6")
+
+	// Search using the Retrokit Platform name
+	out, _, err = cli.Run("core", "search", "rk002", "--all", "--additional-urls="+url.String(), "--format", "json")
+	require.NoError(t, err)
+	checkPlatformIsInJSONOutput(out, "Retrokits-RK002:arm", "1.0.5")
+	checkPlatformIsInJSONOutput(out, "Retrokits-RK002:arm", "1.0.6")
+
+	// Search using board names
+	out, _, err = cli.Run("core", "search", "myboard", "--all", "--additional-urls="+url.String(), "--format", "json")
+	require.NoError(t, err)
+	checkPlatformIsInJSONOutput(out, "Package:x86", "1.2.3")
+
+	// Check search with case, accents and spaces
```

Comment:
I recommend moving this information to comments at the specific test cases below to make it clear what their intended purpose is.